modular_skyrat/modules/aesthetics/windows/code/windows.dm

```diff
@@ -25,3 +25,47 @@
 	canSmoothWith = list(SMOOTH_GROUP_WALLS, SMOOTH_GROUP_WINDOW_FULLTILE, SMOOTH_GROUP_AIRLOCK)
 
 /obj/structure/window/reinforced/fulltile/ice
+
+// Polarized Windows
+
+/// First defines and allows it to be assigned an ID to be toggled
+/obj/structure/window/reinforced/fulltile/polarized
+	name = "Polarized Window"
+	desc = "Adjusts its tint with voltage, it is reinforced with metal rods."
+	/// ID field check, allow buttons/switches etc to call an ID and toggle window
+	var/id = null
+
+/// Toggles the tingint effect for the window
+/obj/structure/window/reinforced/fulltile/polarized/proc/toggle()
+	if(opacity)
+		animate(src, color="#FFFFFF", time=5)
+		set_opacity(FALSE)
+		alpha = initial(alpha)
+	else
+		animate(src, alpha = 255, color="#222222", time=5)
+		set_opacity(TRUE)
+
+/obj/machinery/button/window/reinforced/polarized
+	name = "Polarized Window Button"
+	desc = "A remote control switch for polarized windows."
+	/// Checks the state and assigns it to default FALSE - toggling purposes
+	var/active = FALSE
+
+/obj/machinery/button/window/reinforced/polarized/attack_hand(mob/living/user, list/modifiers)
+	if(..())
+		return TRUE
+
+	toggle_tint()
+/// Enables the button to check and ensure it's both active and matching the ID
+/obj/machinery/button/window/reinforced/polarized/proc/toggle_tint()
+    active = !active
+	/// ID Check before running the toggle proc
+    for(var/obj/structure/window/reinforced/fulltile/polarized/Window)
```

Comment:
i can't remember if this'll do nothing or do an implicit `in world`, either way use a global list of polarized windows instead of this

Reply:
global?

Reply:
after i removed the extras after the var assignment  it started animating properly - windows dont really have any proc's on them so i couldnt piggyback and GLOB.structure didnt exist

Reply:
make a new GLOB, GLOB.polarized_windows or smth, add windows to it on initialize and remove on destroy